### PR TITLE
Correct double slashes within the path as well

### PR DESCRIPTION
**Pull request by sophiadavis**

### CODE REVIEW

File: byterestclient/__init__.py [hunk 1]
```diff
@@ -37,8 +38,11 @@ def request(self, method, path, data=None, *args, **kwargs):
         else:
             return response.json
 
-    def get_absolute_url(self, path):
-        return self.endpoint.rstrip('/') + '/' + path.lstrip('/')
+    def format_absolute_url(self, path):
+        url = urlsplit(self.endpoint)
+        total_path = '/'.join([p.lstrip('/') for p in [url.path, path]])
+        clean_path = total_path.replace('//', '/')
+        return urlunsplit([url.scheme, url.netloc, clean_path, url.query, url.fragment])
```
Comment (allardhoeve) [inline byterestclient/__init__.py]: maybe we can simplify it using a regex

**Comment by allardhoeve** [comment]
@mvschaik

File: byterestclient/__init__.py (hunk 1 above)
Comment (allardhoeve) [inline byterestclient/__init__.py]: or just find a more easily-read version of line 53 :smile:

**Comment by allardhoeve** [comment]
For the record: this tries to help you when you've accidentally written requests where you concat the endpoint with / at the end with a url part that starts with one. For example when the endpoint is configured in hiera and the path is configured in code.

**Comment by mvschaik** [comment]
Isn't it easier to use something like [urlparse.urljoin](https://docs.python.org/2/library/urlparse.html#urlparse.urljoin) or maybe even [os.path.join](https://docs.python.org/2/library/os.path.html#os.path.join) if `os.sep` is `/`?

**Comment by allardhoeve** [comment]
```
# note that this removes /noot/ !
In [11]: urlparse.urljoin('http://aap.nl/noot/', '/mies/')
Out[11]: 'http://aap.nl/mies/'

# note that this also removes /noot !
In [16]: urlparse.urljoin('http://aap.nl//noot', 'mies/')
Out[16]: 'http://aap.nl//mies/'

# what we want.
In [12]: urlparse.urljoin('http://aap.nl/noot/', 'mies/')
Out[12]: 'http://aap.nl/noot/mies/'
```

So this does not do exactly what we want.

**Comment by allardhoeve** [comment]
So, this function does two things, one of them is new:

1. Add path component to endpoint.
2. Fix any double-slashes for the user. (new)

Using urlunsplit makes us lose important parts of the URL, for example the port, username and password. We will probably not use that often, but it is not really concise.

https://docs.python.org/2/library/urlparse.html#urlparse.urlsplit

We should use `urlunparse`, which takes all the things into account.

**Comment by allardhoeve** [comment]
Our test should properly re-assemble this string, then there is no more doubt :smile:

```
In [44]: urlparse.urlparse('http://henk:pass@aap.nl:8080/mies/henk;zus?aap=noot#noot')
Out[44]: ParseResult(scheme='http', netloc='henk:pass@aap.nl:8080', path='/mies/henk', params='zus', query='aap=noot', fragment='noot')
```

**Comment by allardhoeve** [comment]
```python
def format_absolute_url(self, path):
    def join_and_clean_paths(path1, path2):
        return re.sub(r'/+', '/', path1 + '/' + path2

    urlparts = list(urlsplit(self.endpoint))  # cast to list, because resulting tuple is immutable
    urlparts[2] = join_and_clean_paths(urlparts[2], path)
    return urlunsplit(urlparts)
```

**Comment by allardhoeve** [comment]
almost there

**Comment by sophiadavis** [comment]
@allardhoeve Are we there yet?

**Comment by allardhoeve** [comment]
nice

**Pull request merged**